plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -247,6 +251,16 @@ public function get_current_orders_pending_sync_count( $use_cache = false ): int
 
 		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
 		$pending_count = (int) $wpdb->get_var( $sql );
+
+		$deleted_from_table = $this->custom_orders_table_is_authoritative() ? $orders_table : $wpdb->posts;
```

Comment:
Isn't `posts` a property in the `$wpdb` object that returns the name of the posts table?

Or you mean `$orders_table`? Earlier in the method we're doing `$orders_table  = $this->data_store::get_orders_table_name();` which is the proper thing to do, right?

Reply:
> Or you mean $orders_table? Earlier in the method we're doing $orders_table  = $this->data_store::get_orders_table_name(); which is the proper thing to do, right?

It is proper when we are fetching data, but I don't think that it's safe to store the exact table names, since the wpdb prefix can change in the future for a shop. If it changes, these records will be invalided.
However, if we store a constant instead, like `posts` or `orders` than it will be isolated from this change.

Reply:
Oh you mean the actual meta value that gets stored. Ok, that makes sense, I'll change it.

Reply:
Implemented in https://github.com/woocommerce/woocommerce/pull/37050/commits/f589ecfd38c55ba7b44cee8ec43bab48171a6984 (and pull request description updated).